submissions/m-ruslan/memory-pair-game/app.js

```diff
@@ -0,0 +1,135 @@
+// main logic of game
+document.addEventListener("DOMContentLoaded", () => {
+  let cardID, cardValue, prevCardId, prevCardValue;
+  let qOfOpenCards = 0;
+
+  startNewGame();
+
+  document.querySelector("main").addEventListener("click", (event) => {
+    let aEl = event.target.closest(".flip-container");
+    if (!aEl) return;
+    if (aEl.id === prevCardId && aEl.id !== undefined) return;
+
+    if (qOfOpenCards < 2) {
+      qOfOpenCards++;
+      cardID = aEl.id;
+      cardValue = aEl.getAttribute("value");
+
+      flip(cardID);
+
+      if (qOfOpenCards === 1) {
+        prevCardId = cardID;
+        prevCardValue = cardValue;
+      } else if (qOfOpenCards === 2) {
+        setTimeout(() => {
+          if (cardValue === prevCardValue && cardID !== prevCardId) {
+            match(cardID, prevCardId);
+          } else {
+            dontMatch(cardID, prevCardId);
+          }
+
+          qOfOpenCards = 0;
+          prevCardId = null;
+          prevValue = null;
+
+          if (qOfCardsLeft === 0) {
+            youWon();
+          }
+        }, 1000);
+      }
+    }
+  });
+});
+
+//main functions
+const startNewGame = () => {
+  clearField();
+  createField(arrOfCards);
+  qOfCardsLeft = arrOfCards.length;
+};
+
+const youWon = () => {
+  alert("You Won!");
+  startNewGame();
+};
+
+const clearField = () => {
+  document.querySelector("main").innerHTML = "";
+};
+
+const createField = (arr) => {
+  let shuffledArr = shuffle(arr);
+  let cardsfield = document.createDocumentFragment();
+  for (let i = 0; i < shuffledArr.length; i++) {
+    let card = document.createElement("div");
+    card.innerHTML = createCard(shuffledArr[i], i);
+    cardsfield.appendChild(card);
+  }
+  document.querySelector("main").appendChild(cardsfield);
+};
+
+const shuffle = (arr) => {
+  return arr.sort(() => 0.5 - Math.random());
+};
+
+const createCard = (content, index) => {
+  return `<div class="flip-container" value="${content.value}" id="card-${index}">
+        <div class="flipper">
+            <div class="front"></div>
+            <div class="back">
+            <img src="${content.src}" alt="${content.alt}">
+            </div>
+        </div>
+    </div>`;
+};
+
+const flip = (cardID) => {
+  document.querySelector("#" + cardID).classList.toggle("flip");
+};
+
+const match = (card1ID, card2ID) => {
```

Comment:
function must be verbs and show what it will do. for example in this case `hideMatchedCards` is more corresponding naming